Change lock state is enum (#2329)
home-assistant/developers.home-assistant: blog/2024-09-23-state-constants-lock-deprecation.md

---
author: G Johansson
authorURL: https://github.com/gjohansson-ST
authorImageURL: https://avatars.githubusercontent.com/u/62932417?v=4
authorTwitter: GJohansson
title: "Deprecating state constants for lock"
---

> As of Home Assistant Core 2024.10, the constants used as returning state in `LockEntity` is deprecated and replaced by the `LockState` enum.

Reviewer: ```suggestion
As of Home Assistant Core 2024.10, the constants used for returning state in `LockEntity` are deprecated and replaced by the `LockState` enum.
```

> As of Home Assistant Core 2024.10, the constants used as returning state in `LockEntity` is deprecated and replaced by the `LockState` enum.

Author: ```suggestion
As of Home Assistant Core 2024.10, the constants used to return state in `LockEntity` are deprecated and replaced by the `LockState` enum.
```

There is a one-year deprecation period, and the constants will stop working from 2025.10 to ensure all custom integration authors have time to adjust.

> As the `state` property is not meant to be overwritten, in most cases this is only related for use in other properties or in tests rather in a `state` property.

Reviewer: ```suggestion
As the `state` property is not meant to be overwritten, in most cases this change will only affect other Entity properties or tests rather than the `state` property.
```
